ballerina/executor_visitor.bal

```diff
@@ -15,76 +15,182 @@
 // under the License.
 
 import graphql.parser;
+import ballerina/jballerina.java;
+import ballerina/log;
 
-class ExecutorVisitor {
+isolated class ExecutorVisitor {
     *parser:Visitor;
 
-    private final __Schema schema;
+    private final readonly & __Schema schema;
     private final Engine engine; // This field needed to be accessed from the native code
     private Data data;
     private ErrorDetail[] errors;
     private Context context;
-    private any|error result;
+    private any|error result; // The value of this field is set from the native code
 
     isolated function init(Engine engine, __Schema schema, Context context, any|error result = ()) {
         self.engine = engine;
-        self.schema = schema;
+        self.schema = schema.cloneReadOnly();
```

Comment:
Since the `Engine` class has `__Schema & readonly` we can make the parameter type to `readonly & __Schema` and avoid the cloning, right?

Reply:
fixed